src/main/java/matveyeva/phonebook/crud/UserCRUD.java

```diff
@@ -0,0 +1,151 @@
+package matveyeva.phonebook.crud;
+
+import matveyeva.phonebook.entity.User;
+import matveyeva.phonebook.exception.InvalidUserException;
+import org.apache.log4j.Logger;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+public class UserCRUD {
+
+    private static Set<User> users = new HashSet<User>();
+    private static Logger logger;
+    private ContactCRUD contactCRUD;
+
+    public UserCRUD(){
+        logger = Logger.getLogger(this.getClass());
+        try {
+            loadUsers();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public User createUser(String str){
+        try{
+            User user = split(str);
+            users.add(user);
+            logger.info("New user " + user.getUserName() + " created");
+            return user;
+        }catch (InvalidUserException ex){
+            System.out.println(ex.getMessage());
+            return null;
+        }catch (IllegalArgumentException iex){
+            System.out.println("User exists");
+            return null;
+        }
+
+//        if(user != null && user.isUserValid()){
+//            try {
+//                if(users.add(user)){
+//                    logger.info("New user " + user.getUserName() + " created");
+//
+//                    return user;
+//                }
+//            } catch (Exception e) {
+//                e.printStackTrace();
+//            }
+//        }
+//        return null;
+    }
+
+    public void delete(User user) throws IOException {
+        contactCRUD = new ContactCRUD(user);
+        contactCRUD.deleteAll();
+        try {
+            contactCRUD.reloadContacts();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        users.remove(user);
+        reloadUsers();
+    }
+
+    public void deleteAll() throws IOException {
+        for(User user: users){
+            contactCRUD = new ContactCRUD(user);
+            contactCRUD.deleteAll();
+        }
+        contactCRUD.reloadContacts();
+        ArrayList<User> arr = new ArrayList<User>(users);
+        users.removeAll(arr);
+        reloadUsers();
+    }
+
+    public User update(String newUser, User oldUser){
+        try{
+            User user = split(newUser);
+            users.remove(oldUser);
+            users.add(user);
+            return user;
+        }catch(InvalidUserException ex){
+            System.out.println(ex.getMessage());
+            return null;
+        }
+    }
```

Comment:
In this case updating is ok but if we have more than 2 fileds for User entity we shouldn't remove full object from cache we should update them.